### DISCUSSION TITLE
Manipulating materials with Slide3D editor

### DISCUSSION THREAD

**Discussion by meirlebel** [google/model-viewer#1837] [Ideas]
Slide3D  (www.slide3d.com) is a web site for creating 3D web presentations using model-viewer technology.

https://user-images.githubusercontent.com/38201612/104847711-785ec380-58ea-11eb-9a17-50849ae029c3.mp4

model-viewer team, do you plan to add methods like:

materialFromPoint(pixelX, pixelY);
objFromPoint(pixelX, pixelY); or meshFromPoint(pixelX, pixelY);

Like the existing positionAndNormalFromPoint(pixelX, pixelY)?

Tn'x
Meir Lebel
www.slide3d.com

**Comment by elalish** [comment]
Yes, I think those are good ideas, though they'll happen sooner if someone is willing to contribute. I think materialFromPoint would be pretty straight-forward now, but for meshFromPoint we'll need to first expose the list of glTF meshes in the scene-graph API, which will also be important for things like hiding/showing nodes: https://github.com/google/model-viewer/discussions/1817
